### PR TITLE
Base configuration for the plugin

### PR DESCRIPTION
**Pull request by Tabrisrp**
- Add composer and PHP CS configuration
- Add event manager and container
- Add base plugin files

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	 * Loads the plugin in WordPress
+	 *
+	 * @return void
+	 */
+	public function load() {
+		if ( $this->is_loaded() ) {
+			return;
+		}
+
+		$container = new Container();
```
Comment (engahmeds3ed) [inline src/Plugin.php:36]: don't we need to add a filter here so we can use on support helper plugins? like what we do at WPR plugin `rocket_container`? what do u think?

**Review by engahmeds3ed — CHANGES_REQUESTED** [review]

File: rocketcdn.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+defined( 'ABSPATH' ) || exit;
+
+require realpath( plugin_dir_path( __FILE__ ) ) . '/includes/RocketCDNRequirementsCheck.php';
+
+$rocketcdn_rq_check = new RocketCDNRequirementsCheck(
+	[
+		'plugin_name'    => 'RocketCDN',
+		'plugin_version' => '1.0',
+		'wp_version'     => '5.4',
+		'php_version'    => '7.0',
```
[the remark this answers is not in the document]
Comment (Tabrisrp) [reply rocketcdn.php:25]: those are not used anywhere else, I'm not sure there is much value in changing that for now

File: rocketcdn.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * Author: WP Media
+ * Author URI: https://wp-media.me
+ * Licence: GPLv3 or later
+ *
+ * Copyright 2021 WP Rocket
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+require realpath( plugin_dir_path( __FILE__ ) ) . '/includes/RocketCDNRequirementsCheck.php';
```
[the remark this answers is not in the document]
Comment (Tabrisrp) [reply rocketcdn.php:18]: In this case it doesn't make a difference, the path is automatically converted by PHP as needed

File: src/Plugin.php (hunk 1 above)
Comment (Tabrisrp) [reply src/Plugin.php]: Updated the code for that 👍🏼

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	},
+	"require": {
+		"php": ">=7.0",
+		"composer/installers": "~1.0",
+		"psr/container": "1.0.0"
+	},
+	"require-dev": {
+		"php": "^7 || ^8",
+		"brain/monkey": "^2.0",
+        "coenjacobs/mozart": "^0.7",
```
[the remark this answers is not in the document]
Comment (Tabrisrp) [reply composer.json:33]: Should be fixed now

**Review by engahmeds3ed — APPROVED** [review]

**Pull request merged**